src/editors/data/redux/problem/reducers.js

```diff
@@ -41,8 +44,62 @@ const problem = createSlice({
     }),
     updateQuestion: (state, { payload }) => ({
       ...state,
-      question: payload
+      question: payload,
     }),
+    updateAnswer: (state, { payload }) => {
+      // you can mutuate state only inside creating
+      // https://redux-toolkit.js.org/usage/immer-reducers#immutable-updates-with-immer
+      const { id, hasSingleAnswer, ...answer } = payload;
+      state.answers = state.answers.map(obj => {
+        if (obj.id === id) {
+          return { ...obj, ...answer };
+        }
+        // set other answers as incorrect if problem only has one answer correct
+        // and changes object include correct key change
+        if (hasSingleAnswer && has(answer, 'correct') && obj.correct) {
+          return { ...obj, correct: false };
+        }
+        return obj;
+      });
+    },
+    deleteAnswer: (state, { payload }) => {
+      const { id } = payload;
+      if (state.answers.length <= 1) {
+        return state;
+      }
+      state.answers = state.answers.filter(obj => obj.id !== id).map((answer, index) => {
+        const newId = indexToLetterMap[index];
+        if (answer.id === newId) {
+          return answer;
+        }
+        return { ...answer, id: newId };
+      });
+    },
+    addAnswer: (state) => {
+      const currAnswers = state.answers;
+      if (currAnswers.length >= indexToLetterMap.length) {
+        return state;
+      }
+      const [firstAnswer] = currAnswers;
+      const newOption = {
+        id: currAnswers.length ? nextAlphaId(currAnswers[currAnswers.length - 1].id) : 'A',
+        title: '',
+        selectedFeedback: undefined,
+        unselectedFeedback: undefined,
+        feedback: undefined,
+        correct: false,
+      };
+      if (isUndefined(firstAnswer.feedback)) {
+        newOption.selectedFeedback = '';
+        newOption.unselectedFeedback = '';
+      } else {
+        newOption.feedback = '';
+      }
+      state.answers = [
```

Comment:
any reason we aren't returning  a value for this function?

Reply:
Will fix this :) it should return a value